common/djangoapps/student/models.py

```diff
@@ -1605,7 +1611,13 @@ def enrollments_for_user_with_overviews_preload(cls, user):  # pylint: disable=i
         The name of this method is long, but was the end result of hashing out a
         number of alternatives, so pylint can stuff it (disable=invalid-name)
         """
-        enrollments = list(cls.enrollments_for_user(user))
+
+        enrollments = cls.enrollments_for_user(user)
+        if not load_all_courses:
+            courses_count = cls.get_dashboard_course_limit()
+            if courses_count:
+                enrollments = cls.enrollments_for_user(user).order_by('user__id', '-created')[:courses_count]
```

Comment:
As a general tip: When you use `user__id` (with two underscores), it will join the relation back to user, and use `auth_user.id`. But you don't need to do that join to sort by something. Using `user_id` will get you the same effect without the join.

That being said, in this case the user_id is already constrained by the `filter` in `enrollments_for_user`, so it shouldn't be necessary. Just doing `.order_by('-created')` should do the right thing. But definitely worth testing.